Scalar.Common/Platforms/Mac/MacPlatform.cs

```diff
@@ -171,5 +184,43 @@ public override string ScalarBinDirectoryName
             // Documented here (in the addressing section): https://www.unix.com/man-page/mojave/4/unix/
             public override int MaxPipePathLength => 104;
         }
+
+        private static class NativeMethods
+        {
+            // Definitions from
+            // /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk
+            // /stdlib.h
+
+            [DllImport("libc", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
+            private static extern IntPtr realpath([In] IntPtr file_name, [In, Out] IntPtr resolved_name);
+
+            public static string ResolveSymlink(string path)
+            {
+                const int PATH_MAX = 0;
+
+                IntPtr pathBuf = IntPtr.Zero;
+                IntPtr resolvedBuf = IntPtr.Zero;
+
+                try
+                {
+                    pathBuf = Marshal.StringToHGlobalAuto(path);
+                    resolvedBuf = Marshal.AllocHGlobal(PATH_MAX);
```

Comment:
SuperNit: maybe PATH_MAX + 1 ??

Reply:
The docs state:
> The realpath() function resolves all symbolic links, extra ``/'' charac-
     ters, and references to /./ and /../ in file_name, and copies the result-
     ing absolute pathname into the memory referenced by resolved_name.  The
     resolved_name argument must refer to a buffer capable of storing at least
     PATH_MAX characters.

So _at least_ would be `== PATH_MAX`, but for good measure I'll `+1`.

Reply:
..also no one noticed that I had left `const int PATH_MAX = 0;`!!! 🤦

Reply:
Heh. I did miss that. But the `PATH_MAX` characters really amounts to `PATH_MAX + 1` bytes, as the `NUL` at the end is not a character.